framework/src/org/checkerframework/common/value/ValueCheckerUtils.java

```diff
@@ -97,7 +98,8 @@
         } else if (AnnotationUtils.areSameByClass(anno, BottomVal.class)) {
             values = convertBottomVal(anno, castType);
         } else if (AnnotationUtils.areSameByClass(anno, UnknownVal.class)
-                || AnnotationUtils.areSameByClass(anno, ArrayLen.class)) {
+                || AnnotationUtils.areSameByClass(anno, ArrayLen.class)
+                || AnnotationUtils.areSameByClass(anno, IntRange.class)) {
```

Comment:
Can you explain why "we have to check if `@IntRange` spans 10 or fewer elements in each and every places"?  I don't see that that is necessary.  The checker has to handle operations on mixed types, but it could do so by just converting from `@IntVal` to `@IntRange`.  That's something that must be implemented anyway, so it wouldn't be any extra work.  In the future, converting from small `@IntRange` to `@IntVal` might add precision, but I'm not even sure about that and it could be delayed until we have evidence that the added precision is worth the implementation effort.